starknet/src/voting_strategies/evm_slot_value.cairo

```diff
@@ -91,6 +123,27 @@ mod EvmSlotValueVotingStrategy {
         }
     }
 
+    #[generate_trait]
+    impl InternalImpl of InternalTrait {
+        fn get_mapping_slot_key(mapping_key: u256, slot_index: u256, offset: u32) -> u256 {
+            keccak::keccak_u256s_be_inputs(
+                array![
+                    keccak::keccak_u256s_be_inputs(array![mapping_key, slot_index].span())
+                        .byte_reverse()
+                ]
+                    .span()
+            )
+                .byte_reverse()
+                + integer::U32IntoU256::into(offset)
+        }
+
+        fn decode_checkpoint_slot(slot: u256) -> (u32, u256) {
+            let block_number = slot.low & 0xffffffff;
```

Comment:
could be nice to explain why it's on the first 4 bytes, maybe link to the checkpoint struct somewhere?

Reply:
added a detailed comment